src/sentry/replaystore/base.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import abc
+from dataclasses import dataclass
+from datetime import datetime
+from enum import IntEnum
+from threading import local
+from typing import Any, Dict, List, Tuple, cast
+
+from sentry.utils import json
+from sentry.utils.services import Service
+from sentry.utils.strings import decompress
+
+
+class ReplayNotFound(ValueError):
+    pass
+
+
+class ReplayDataType(IntEnum):
+    INIT = 1
+    EVENT = 2
+    PAYLOAD = 3
+
+    @staticmethod
+    def should_be_encoded() -> list[ReplayDataType]:
```

Comment:
This name makes me think that it will return a bool. Could we call this `encoded_types`?